app/models/nic/base.rb

```diff
@@ -148,6 +151,13 @@ def host_managed?
 
     protected
 
+    def compute_resource_requires_fields
+      if host.compute_resource.present?
```

Comment:
I don't think it needs to be in interfaces, or touch interfaces at all - it could be on the host, when the compute resource is specified, but no (inherited) compute profile or attributes are provided, then file validation issues there.  It doesn't have a lot to do with NICs.